src/rpc/blockchain.cpp

```diff
@@ -1516,8 +1512,14 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
         }
     }
 
+    blockcount = std::min(blockcount, pindex->nHeight - 1);
```

Comment:
I wonder if the error could be avoided by making the algorithm compatible with 1 block chain.